src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -474,42 +371,56 @@ void spacetime_deriv_damped_harmonic_h(
   }
 }
 
-// Explicit Instantiations
-/// \cond
+template <size_t SpatialDim, typename Frame>
+void DampedHarmonicCompute<SpatialDim, Frame>::function(
+    const gsl::not_null<return_type*> h_and_d4_h,
+    const db::item_type<Tags::InitialGaugeH<SpatialDim, Frame>>& gauge_h_init,
+    const db::item_type<Tags::SpacetimeDerivInitialGaugeH<SpatialDim, Frame>>&
+        dgauge_h_init,
+    const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, SpatialDim, Frame>& shift,
+    const tnsr::a<DataVector, SpatialDim, Frame>&
+        spacetime_unit_normal_one_form,
+    const Scalar<DataVector>& sqrt_det_spatial_metric,
+    const tnsr::II<DataVector, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::aa<DataVector, SpatialDim, Frame>& spacetime_metric,
+    const tnsr::aa<DataVector, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataVector, SpatialDim, Frame>& phi, const double time,
+    const double t_start, const double sigma_t,
+    const tnsr::I<DataVector, SpatialDim, Frame>& coords,
+    const double sigma_r) noexcept {
+  if (UNLIKELY(h_and_d4_h->number_of_grid_points() != get(lapse).size())) {
```

Comment:
I'm curious exactly what UNLIKELY() actually does...

Reply:
Great question! When the compiler reaches an `if` it has the choice of making evaluating the `true` or `false` branch the default. CPUs use this to precompute things inside the `true` or `false` branch, through away the result if the other branch was taken. The idea is rather than waiting for the conditional to finish before doing more work, the CPU can continue to do work, then just throw it away if it wasn't the right work. Thus, in the case that work was thrown away you're no slower than you would've been had you waited, in the case where you predicted the right branch you are faster than you would've been had you waited. Since the best the compiler can do is randomly choose either the `true` or `false` branch as being the one that will be evaluated most often you basically get a 50/50 chance of seeing speedups (the hardware does some memorization to try and improve this). However, if we, the programmer, know that 99% of the time we will take the `false` branch, by giving that info to the compiler we can see some performance gains. While overall in practice I don't expect any individual ones to make a huge impact, it's an easy thing to do that I find also helps signal some information to other developers on how the code is expected to behave. Does that make sense?

Reply:
That’s very cool. Thanks!